### PR TITLE
Makes generatecss.php also generate .min.css

### PR DESCRIPTION
**Pull request by ghost**
Github: https://github.com/joomla/joomla-cms/pull/6979
JIssues: http://issues.joomla.org/tracker/joomla-cms/6979

In this PR:
- Made the generatecss.php also generate .min.css files
- Tweaked the generatecss.php to improve output
- Re-compiled the css files
- Added the .min.css files
- Made the templates/files use the change the.min.css files (except when in DEBUG mode)

Changes I made to the generatecss.php:
- Made the function style values have spaces after commas, like `rgb(2555, 255, 255, 1)`
- Expands # rgb colours to the full 6 digit format, so `#fc9` becomes `#ffcc99`

### CODE REVIEW

File: libraries/cms/less/formatter/joomla.php [hunk 1]
```diff
@@ -18,13 +18,57 @@
...
+}
+
+/**
+ * Formatter ruleset for Joomla formatted CSS (minified) generated via LESS
+ *
+ * @package     Joomla.Libraries
+ * @subpackage  Less
+ * @since       3.4.2
+ */
+class JLessFormatterJoomlaCompressed extends lessc_formatter_classic
```
Comment (wilsonge) [inline libraries/cms/less/formatter/joomla.php]: Can we just call this ```JLessFormatterCompressed``` and put it in a new file for autoloading please

Comment (nonumber) [inline libraries/cms/less/formatter/joomla.php]: I don't know. Can we?
So have a libraries/cms/less/formatter/compressed.php ?

What about the JLessFormatterJoomla ?
Does that name (and file name) make sense?
Or should it be something like JLessFormatterDefault (and default.php) ?
And have JLessFormatterJoomla extend JLessFormatterDefault for B/C...

Comment (wilsonge) [inline libraries/cms/less/formatter/joomla.php]: I can't see why not. So yeah exactly that. I personally am not a huge fan of the name ```Joomla``` the ```JLessFormatterDefault``` one is nicer imo.

Comment (nonumber) [inline libraries/cms/less/formatter/joomla.php]: Done :)

File: libraries/cms/less/formatter/joomla.php [hunk 2]
```diff
@@ -9,22 +9,16 @@
 
 defined('JPATH_PLATFORM') or die;
 
+require_once __DIR__ . '/default.php';
```
Comment (wilsonge) [inline libraries/cms/less/formatter/joomla.php]: Packing up from Prague so can't check right now but shouldn't this class be autoloaded?

Comment (nonumber) [inline libraries/cms/less/formatter/joomla.php]: Yep, removing line...

**Comment by nonumber** [comment]
Closing this because I don't have the time and will to keep updating this to fix merge conflicts and then have to do it all over again because nobody tests it for months.

**Pull request closed without merging**